routes/runFunction.go

```diff
@@ -0,0 +1,44 @@
+package routes
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+
+	"github.com/do4-2022/grobuzin/database"
+)
+
+func (c *Controller) RunFunction(ctx *gin.Context) {
+    fnID, err := uuid.Parse(ctx.Param("id"))
+	
+	var fnBody any; // because the body is entirely defined by the user, we just forward it to the function
+
+	if err := ctx.BindJSON(&fnBody); err != nil {
+		ctx.JSON(400, gin.H{"error": "Could not parse function body"})
+		return
+	}
+
+	if err != nil {
+		ctx.JSON(400, gin.H{"error": "Invalid function ID"})
+		return
+	}
+
+	var fn database.FunctionState
+	tx := c.DB.Where(&database.FunctionState{ID: fnID}).First(&fn)
+
+
+	// put these two into gofuncs
+	if tx.Error != nil {
+		// todo
+		// request a function to the scheduler here
+		// fn = something
+	}
+
+	if fn.Status == "Pending" {
+		// scheduler.WaitForReadiness(fn.Id)
+		// Maybe add a pubsub in the PG that triggers when a function is ready aka when the status is updated, https://www.postgresql.org/docs/current/sql-notify.html
```

Comment:
@nponsard @alexis-langlet what do you think ?

Reply:
well, for now I'm not sure implementing a pubsub is a great idea.
Let's begin with simple stuff and send a request to check readiness every X ms at first
We will improve that if needed and if time allows it

Reply:
I second this